### PR TITLE
Add go test coverage

### LINKED ISSUE #16: Go (go test coverage)

### PR DESCRIPTION
**Pull request by mr-u0b0dy**
## Summary by Sourcery

Add native Go coverprofile parsing and improve Go-focused coverage path resolution and merging across Cobertura and LCOV formats.

New Features:
- Introduce a Go coverprofile parser to ingest native `go test -coverprofile` reports without conversion.
- Add Go coverage examples with Makefile workflows and sample Go code demonstrating native, LCOV, and Cobertura generation.

Enhancements:
- Improve Cobertura XML parsing to better resolve Go module-qualified paths, honor `<source>` roots, and merge duplicate class and line entries without double-counting.
- Enhance LCOV parsing to resolve Go module-qualified and project-prefixed paths, deduplicate DA line records, and preserve function coverage data.
- Extend coverage file detection, language patterns, and project root markers to recognize Go projects and Go coverprofiles.

Documentation:
- Document Go coverprofile support and usage in the main README and formats reference, including generation and loading instructions.
- Update coverage examples documentation to include the new Go example project and supported tools.

Tests:
- Add parser tests for Go coverprofiles, including valid, empty, and malformed profiles, and extend Cobertura and LCOV specs to cover Go-specific path resolution and line-merging behaviors.

## Summary by CodeRabbit

* **New Features**
  * Native support for Go coverprofile (.out) inputs with detection and a dedicated parser plus Go-aware project resolution.

* **Documentation**
  * Added Go coverage examples, quick-starts, format docs, and Makefile-based workflows for generating/converting reports.

* **Improvements**
  * Enhanced LCOV/Cobertura path resolution, merging/aggregation of duplicate lines/branches, and coverage format detection heuristics.

* **Tests**
  * Added parser and detection tests, Go fixtures, and strengthened test helpers for coverage file creation.

### CODE REVIEW

File: lua/crazy-coverage/utils.lua [hunk 1]
```diff
@@ -47,8 +47,35 @@ end
...
+  end
+
+  return false
+end
+
 function M.detect_format(file_path)
   local ext = file_path:match("%.([^.]+)$")
+  local lines = M.read_file(file_path)
+
+  if is_go_coverprofile(lines) then
```
[the remark this answers is not in the document]
Comment (mr-u0b0dy) [reply lua/crazy-coverage/utils.lua]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/mr-u0b0dy/crazy-coverage.nvim/pull/17#discussion_r2969877715)

File: test/helpers.lua [hunk 2]
```diff
@@ -15,9 +15,9 @@ function M.create_temp_coverage_file(format, content)
   if file then
     file:write(content)
     file:close()
-    return filename, tmp_dir
   end
-  return nil
+
+  return filename, tmp_dir
```
[the remark this answers is not in the document]
Comment (mr-u0b0dy) [reply test/helpers.lua:21]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/mr-u0b0dy/crazy-coverage.nvim/pull/17#discussion_r2969877724)

File: lua/crazy-coverage/parser/lcov.lua [hunk 3]
```diff
@@ -105,109 +377,70 @@ function M.parse(file_path, project_root)
   
   -- Project root can be used for context, but coverage file directory is the base
   if not project_root then
-    project_root = vim.fn.fnamemodify(file_path, ":p:h:h")
+    -- Prefer inferred module root from coverage location before generic fallbacks.
+    project_root = find_go_module_root(coverage_dir, 8)
+      or vim.fn.fnamemodify(file_path, ":p:h:h:h")
+  end
```
[the remark this answers is not in the document]
Comment (mr-u0b0dy) [reply lua/crazy-coverage/parser/lcov.lua:402]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/mr-u0b0dy/crazy-coverage.nvim/pull/17#discussion_r2969956333)

File: coverage-examples/README.md [hunk 4]
```diff
@@ -19,6 +19,13 @@ coverage-examples/
     ├── main.cpp
     ├── math_utils.cpp
     └── math_utils.hpp
+
+└── go/                   # Go examples
+  ├── Makefile          # Go coverage and conversion targets
```
[the remark this answers is not in the document]
Comment (mr-u0b0dy) [reply coverage-examples/README.md]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/mr-u0b0dy/crazy-coverage.nvim/pull/17#discussion_r2970008873)
